dune_client/client.py

```diff
@@ -238,3 +252,125 @@ def refresh_into_dataframe(
             ) from exc
         data = self.refresh_csv(query, performance=performance).data
         return pandas.read_csv(data)
+
+    # CRUD Operations: https://dune.com/docs/api/api-reference/edit-queries/
+    def create_query(
+        self,
+        name: str,
+        query_sql: str,
+        params: Optional[list[QueryParameter]] = None,
+        is_private: bool = False,
+    ) -> int:
+        """
+        Creates Dune Query by ID
+        https://dune.com/docs/api/api-reference/edit-queries/create-query/
+        """
+        query_parameters = {
+            "name": name,
+            "query_sql": query_sql,
+            "private": is_private,
+        }
+        if params is not None:
+            query_parameters["parameters"] = [p.to_dict() for p in params]
+        response_json = self._post(
+            route="query/",
+            params=query_parameters,
+        )
+        try:
+            # No need to make a dataclass for this since it's just a boolean.
+            return int(response_json["query_id"])
+        except KeyError as err:
+            raise DuneError(response_json, "create_query Response", err) from err
+
+    def get_query(self, query_id: int) -> DuneQuery:
+        """
+        Retrieves Dune Query by ID
+        https://dune.com/docs/api/api-reference/edit-queries/get-query/
+        """
+        response_json = self._get(route=f"query/{query_id}")
+        return DuneQuery.from_dict(response_json)
+
+    def update_query(  # pylint: disable=too-many-arguments
+        self,
+        query_id: int,
+        name: Optional[str] = None,
+        query_sql: Optional[str] = None,
+        params: Optional[list[QueryParameter]] = None,
+        description: Optional[str] = None,
+        tags: Optional[list[str]] = None,
```

Comment:
Nit: Ideally I would want to set these when just creating a query too.

Reply:
I do not believe their API supports that (at least not according to the docs). Maybe this is something for @msf or someone else in the Dune org.